usmqe/api/tendrlapi/user.py

```diff
@@ -0,0 +1,147 @@
+"""
+tendrl REST API.
+"""
+
+import json
+import requests
+import pytest
+from usmqe.api.tendrlapi.common import TendrlApi
+
+LOGGER = pytest.get_logger("tendrlapi.user", module=True)
+
+
+class ApiUser(TendrlApi):
+    """ Main class for interact with REST API - user.
+    """
+    def users(self, asserts_in=None):
+        """ Get users.
+
+        Name:        "GET_users",
+        Method:      "GET",
+        Pattern:     "users",
+
+        Args:
+            asserts_in: assert values for this call and this method
+        """
+        pattern = "users"
+        request = requests.get(
+            pytest.config.getini("usm_api_url") + pattern,
+            auth=self._auth)
+        self.print_req_info(request)
+        self.check_response(request, asserts_in)
+        if not request.ok:
+            return False
+        defined_keys = {
+            "email",
+            "username",
+            "name",
+            "role"}
+
+        msg = "User {0} should contain: {1}\n\tUser {0} contains: {2}"
+        for item in request.json(encoding='unicode'):
+            user = item["username"]
+            pytest.check(
+                item.keys() == defined_keys,
+                msg.format(user, defined_keys, item.keys))
+        return request.json(encoding='unicode')
+
+    def user_edit(self, username, data, asserts_in=None):
+        """ Edit a single user
+
+        Args:
+            username: name of user that is going to be updated
+            data: dictionary data about user
+                  have to contain: email, username, name, role
+            asserts_in: assert values for this call and this method
+        """
+        pattern = "users/{}".format(username)
+        request = requests.put(
+            pytest.config.getini("usm_api_url") + pattern,
+            json.dumps(data),
+            auth=self._auth)
+        self.print_req_info(request)
+        self.check_response(request, asserts_in)
+        return request.json(encoding='unicode')
+
+    def user_add(self, user_in, asserts_in={
+            "cookies": None,
+            "ok": True,
+            "reason": 'Created',
+            "status": 201}):
```

Comment:
That's a valid point. Moreover it's highly recommended to not use dictionaries and lists as default values for method parameters. If you need such defaults use None instead and populate the parameter in method using `if <parameter> is None <parameter> = <list or dict>` or maybe even better `<parameter> = <parameter> or <list or dict>`.
Sorry, I should notice that before.